### PR TITLE
Add search.maxResults setting

### LINKED ISSUE #8563: Investigate removing Max results limitation for Search and replace results
Investigate removing Max results limitation for Search and replace results

Currently we set max limit to 10000. We need to investigate on lifting this.

### PR DESCRIPTION
**Pull request by rickpmartin**
This PR fixes: #8563 by adding a setting `search.maxResults` which sets the maximum number of search results returned. `null` (empty) can be used to return unlimited results.

Notes:
- The default functionality is unchanged (meaning we still have a default maximum of 10000).
- The original limitation was added due to a performance issue during replace in files: #8529
- #90383 was already closed but it was never really fixed, the fix for that issue only added a warning when the maximum is reached.
- It is worth considering setting the default to unlimited, as other editors do not have this limitation and it is unexpected (there have been numerous issues opened to support this).

Related issues:
- Search only returns first 10,000 items found #90383
- Investigate removing Max results limitation for Search and replace results #8563
- search results have a limit of 10000 items #29120
- Replace in files will stop after 10000 items #51624

Preview:

Below is a preview showing the new setting functionality, and also a replace of ~240,000 lines without issue.

vscode-max-search-results2

### CODE REVIEW

**Review by roblourens — APPROVED** [review]
Thanks for the PR. Want to also change the default to 20,000? 🚀 

I am just still hesitant about blowing up underpowered machines. But I think 20k would be fine and we still have some time to get feedback in insiders before the next release.

**Comment by rickpmartin** [comment]
Sure thing, I've updated the default value to 20000. 

I'm not sure if this is important but there are workflows awaiting approval as they require a maintainer to approve them. Is that something you are able to do?

**Pull request merged**
